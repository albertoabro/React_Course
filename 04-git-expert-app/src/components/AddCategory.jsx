import { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onChange = ({ target }) => { //this is a destructuring of event
        setInputValue(target.value) //resume of event.target.value
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const trimInputValue = inputValue.trim();

        if( trimInputValue.length <= 1) return;

        //setCategories((categories) => [...categories,inputValue]);
        onNewCategory( trimInputValue);
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label='form'>
            <input
                type="text"
                placeholder="Search Gif"
                value={inputValue}
                onChange={onChange} //If function's parameter it's an event, it's possible to omitted
            />
        </form>
    )
}

AddCategory.propTypes ={
    onNewCategory: PropTypes.func.isRequired
}