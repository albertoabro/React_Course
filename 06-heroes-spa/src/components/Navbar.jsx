import { useContext } from "react";
import { Link, NavLink, useNavigate } from "react-router";
import { AuthContext } from "../auth/context/AuthContext";


export const Navbar = () => {

  const navigate = useNavigate();
  const {name = ''} = JSON.parse(localStorage.getItem('user')) || [];

  const {logout} = useContext(AuthContext);

  const onLogout = () => {

    logout();
    
    navigate('/login', { replace: true });
  }
  
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark p-2">
      <Link className="navbar-brand" to="/">
        Heroes
      </Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">
          <NavLink to="/marvel" className={ ({isActive}) => `nav-item nav-link ${ isActive ? 'active' : '' }` } >
            Marvel
          </NavLink>

          <NavLink to="/dc" className={ ({isActive}) => ` nav-item nav-link ${ isActive ? 'active ' : '' }` } >
            DC
          </NavLink>

          <NavLink to="/search" className={ ({isActive}) => ` nav-item nav-link ${ isActive ? 'active ' : '' }` } >
            Search
          </NavLink>
        </div>
      </div>

      <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
        <ul className="navbar-nav ml-auto">
            <span className="nav-item nav-link text-primary"> {name} </span>
            <button className="nav-item nav-link btn" onClick={onLogout}> Logout</button>
        </ul>
      </div>
    </nav>
  );
};
