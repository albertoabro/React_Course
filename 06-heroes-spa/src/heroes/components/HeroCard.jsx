
import PropTypes from 'prop-types'
import { Link } from 'react-router';

const CharactersByHero = ({alter_ego, characters}) => {
  
  return ( alter_ego === characters ) 
    ? <></>
    : <p className='card-text'>{characters}</p>;
}

export const HeroCard = ({
  id, superhero, publisher, alter_ego, first_appearance, characters
}) => {

  const heroImage = `/assets/${id}.jpg`;

  return (
    <div className='col'>
      <div className='card'>
          <div className='row no-gutters animate__animated animate__fadeIn'>
            <div className='col-4'>
              <img src={heroImage} className='card-img' alt={superhero}/>
            </div>

            <div className="col-8">
              <div className="card-body">
                <h5 className="card-title"> {superhero} </h5>
                <p className='card-text'> {alter_ego} </p>
                <CharactersByHero alter_ego={alter_ego} characters={characters}/>
                <p className='card-text'>
                  <small className='text-muted '> {first_appearance} </small>
                </p>

                <Link className='' to={`/hero/${id}`}>
                  More Information... 
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
  )
}


HeroCard.propTypes = {
    id: PropTypes.string.isRequired,
    superhero: PropTypes.string.isRequired,
    publisher: PropTypes.string.isRequired,
    alter_ego: PropTypes.string.isRequired,
    first_appearance: PropTypes.string.isRequired,
    characters: PropTypes.string.isRequired
}

CharactersByHero.propTypes = {
  alter_ego: PropTypes.string.isRequired,
  characters: PropTypes.string.isRequired
}