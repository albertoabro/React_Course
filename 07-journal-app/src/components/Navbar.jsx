import { useDispatch } from "react-redux"
import { AppBar, Grid2, IconButton, Toolbar, Typography } from "@mui/material"
import { LogoutOutlined, MenuOutlined  } from "@mui/icons-material"
import { startLogout } from "../store/auth/thunks";

export const Navbar = ({drawerWidth=240}) => {

    const dispatch = useDispatch();

    const onLogout =() => {
        dispatch( startLogout() );    
    }

  return (
    <AppBar 
        position='fixed'
        sx={{ 
            width: {sm: `calc(100% - ${ drawerWidth }px )`},
            ml: {sm: `${ drawerWidth }px`}
         }}    
    >
    
    <Toolbar>
        <IconButton
            color="inherit"
            edge="start"
            sx={{ mr: 2, display:{sm: 'none'} }}
        >
            <MenuOutlined />
        </IconButton>

        <Grid2 container
            width="100%" 
            alignItems='center'
            justifyContent='space-between'
        >
            <Typography 
                variant="h6" 
                noWrap 
                component='div'
            >
                Journal App
            </Typography>
            <IconButton color="error" onClick={onLogout}>
                <LogoutOutlined />
            </IconButton>
        </Grid2>
    </Toolbar>
    </AppBar>
  )
}
