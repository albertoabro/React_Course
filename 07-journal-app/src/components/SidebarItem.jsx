import { TurnedInNot } from "@mui/icons-material";
import { Grid2, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material";
import { useMemo } from "react";
import { useDispatch } from "react-redux";
import { setActiveNote } from "../store/journal/journalSlice";

export const SidebarItem = ({ note }) => {

    const dispatch = useDispatch() ;

    const { title, body } = note;

    const newTitle = useMemo( () => {
        return title.length > 17
            ? title.substring(0,17) + '...'
            : title;
    }, [ title ])

    const onNoteSelected = () => {
        dispatch( setActiveNote( note ) );
    } 

  return (
    <ListItem disablePadding>
      <ListItemButton onClick={onNoteSelected}>
        <ListItemIcon>
          <TurnedInNot />
        </ListItemIcon>

        <Grid2 container>
          <ListItemText primary={ newTitle } />
          <ListItemText
            secondary={ body }
          />
        </Grid2>
      </ListItemButton>
    </ListItem>
  );
};
