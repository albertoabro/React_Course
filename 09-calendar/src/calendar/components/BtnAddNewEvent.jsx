import { addHours } from "date-fns";
import { useCalendarStore } from "../../hooks/useCalendarStore";
import { useUiStore } from "../../hooks/useUiStore"

export const BtnAddNewEvent = () => {

  const { openDateModal } = useUiStore();
  const { setActiveEvent } = useCalendarStore();

  const handleClickModal = (event) => {

    setActiveEvent({
       title: '',
        note: '',
        start: new Date().toISOString(),
        end: addHours( new Date(), 2 ).toISOString(),
        bgColor: '#fafafa',
        user: {
            _id: '123',
            name: 'Alberto',
        }
    });

    openDateModal();
  };

  return (
    <button 
      className="btn btn-primary fab"
      onClick={ handleClickModal }
    >
        <i className="fas fa-plus"></i>
    </button>
  )
}
