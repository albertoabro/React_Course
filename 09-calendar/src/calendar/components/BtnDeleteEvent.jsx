import { useCalendarStore } from "../../hooks/useCalendarStore";

export const BtnDeleteEvent = () => {

    const { startDeletingEvent, hasEventSelected } = useCalendarStore();

    const handleDeleteEvent = (event) => {
        startDeletingEvent();
    };

  return (
    <button 
      className="btn btn-danger fab-danger"
      onClick={ handleDeleteEvent }
      style={{
        display: hasEventSelected ? '' : 'none'
      }}
    >
        <i className="fas fa-trash-alt"></i>
    </button>
  )
}
