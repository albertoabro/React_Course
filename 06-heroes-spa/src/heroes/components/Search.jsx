import { useLocation, useNavigate } from "react-router";
import queryString from "query-string";
import { useForm } from "../hooks/useForm"
import { HeroCard } from "./HeroCard"
import { getHeroesByName } from "../helpers/getHeroesByName";

export const Search = () => {

  const navigate = useNavigate();
  const location = useLocation();
  const { q = ''} = queryString.parse( location.search );
  const heroes = getHeroesByName(q);

  const showSearch = (q.length === 0);
  const showNotFound = (q.length > 0) && heroes.length === 0;

  const {searchHero, onInputChange} = useForm({
    searchHero: q
  });


  const handleSearchHero = (event) => {
    event.preventDefault();

    navigate(`?q=${ searchHero }`);

  };

    return (
      <>

        <h1>Search</h1>
        <hr/>
        <div className="row">
          <div className="col-5">
            <h4>Searching</h4>
            <hr/>

            <form onSubmit={ handleSearchHero }>
              <input 
              type="text" 
              placeholder="Search"
              className="form-control"
              name="searchHero"
              autoComplete="off"
              value={searchHero}
              onChange={onInputChange}
            />
            </form>
            <button className="btn btn-outline-primary mt-1">
              Search
            </button>
          </div>

          <div className="col-7">
            <h4>Results</h4>
            <hr/>

            <div className="alert alert-primary animate__animated animate__fadeIn" style={{display: showSearch ? '' : 'none' }}>
              Search a Hero
            </div>

            <div className="alert alert-info animate__animated animate__fadeIn" style={{display: showNotFound ? '' : 'none'}}>
              There's a not results with <b>{ q }</b>
            </div>
            
            {
              heroes.map( hero => 
                <HeroCard key={hero.id} {...hero}/>  
            )}
          </div>
        </div>
                     
      </>
    )
  }
  