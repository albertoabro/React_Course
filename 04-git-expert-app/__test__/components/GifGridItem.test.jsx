import { render, screen } from "@testing-library/react"
import { GifGridItem } from "@/components/GifGridItem"

describe('Test in <GifGridItem>', () => {

    const title = 'Call of Duty'
    const url = 'https://test.com/call-of-duty.jpg'
    
    test('should to match with the snapshot', () => {
        const {container} = render(<GifGridItem title={title} url={url} />)
        expect(container).toMatchSnapshot();

    });

    test('should show the image with the URL and ALT indicated', () => {
        render(<GifGridItem title={title} url={url} />)
        // screen.debug();
        // expect (screen.getByRole('img').alt ).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })
    
    test('should show the componentÂ´s title', () => {
        const {container} = render(<GifGridItem title={title} url={url} />)
        expect (screen.getByText(title)).toBeTruthy();
    })
    
    
})
