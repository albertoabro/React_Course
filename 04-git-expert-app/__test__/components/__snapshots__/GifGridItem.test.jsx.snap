// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem> should to match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Call of Duty"
      src="https://test.com/call-of-duty.jpg"
    />
    <p>
      Call of Duty
    </p>
  </div>
</div>
`;
